### PR TITLE
check for java installs in PATH on windows

### PR DESCRIPTION
**Pull request by TayouVR**
this should find java installs from scoop as well as any other installer, that registers java in the PATH environment variable.

No idea if there was an issue for this, I couldn't find one, so I guess I never made one..

### CODE REVIEW

**Review by Scrumplex — APPROVED** [review]
Looks good

File: launcher/java/JavaUtils.cpp [hunk 1]
```diff
@@ -177,6 +177,12 @@ QStringList addJavasFromEnv(QList<QString> javas)
     QByteArray env = qgetenv("POLYMC_JAVA_PATHS");
 #if defined(Q_OS_WIN32)
     QList<QString> javaPaths = QString::fromLocal8Bit(env).replace("\\", "/").split(QLatin1String(";"));
+    
+    QByteArray envPath = qgetenv("PATH");
+    QList<QString> javaPathsfromPath = QString::fromLocal8Bit(envPath).replace("\\", "/").split(QLatin1String(";"));
```
Comment (flowln) [inline launcher/java/JavaUtils.cpp]: From the table in [qEnvironmentVariable's docs](https://doc.qt.io/qt-6/qtglobal.html#qEnvironmentVariable), it seems that we should use that instead here, like:
```cpp
    auto envPath = qEnvironmentVariable("PATH");
    auto javaPathsfromPath = envPath.replace("\\", "/").split(QLatin1String(";"));
```

Comment (Scrumplex) [reply launcher/java/JavaUtils.cpp]: I guess MMC didn't use it, because it targeted Qt 5.6 :skull:

Comment (TayouVR) [reply launcher/java/JavaUtils.cpp]: I will try changing it, but my local builds got some issues. So gotta wait on CI to finish to test if it works correctly.

**Comment by TayouVR** [comment]
tested newest CI build, works! _(also nobody saw the force pushes, I definitely didn't overlook a line which caused builds to fail)_
image

**Review by timoreo22 — APPROVED** [review]

**Review by Scrumplex — APPROVED** [review]

**Review by flowln — APPROVED** [review]
thanks!

File: launcher/java/JavaUtils.cpp (hunk 1 above)
Comment (DioEgizio) [reply launcher/java/JavaUtils.cpp]: > I guess MMC didn't use it, because it targeted Qt 5.6 💀

*still targets 5.4 actually

**Pull request merged**